Tests/kaas/k8s-version-recency/k8s-version-recency-check.py

```diff
@@ -0,0 +1,465 @@
+        if not isinstance(other, K8sVersionInfo):
+            return False
+        return self.major == other.major and self.minor == other.minor and self.patch == other.patch
+
+    def __gt__(self, other):
+        if not isinstance(other, K8sVersionInfo):
+            return False
+        patchcomp = self.minor == other.minor and self.patch > other.patch
+        return self.major > other.major or (self.major == other.major and (self.minor > other.minor or patchcomp))
+
+    def __ge__(self, other):
+        if not isinstance(other, K8sVersionInfo):
+            return False
+        patchcomp = self.minor == other.minor and self.patch >= other.patch
+        return self.major > other.major or (self.major == other.major and (self.minor > other.minor or patchcomp))
+
+    def __lt__(self, other):
+        if not isinstance(other, K8sVersionInfo):
+            return False
+        patchcomp = self.minor == other.minor and self.patch < other.patch
+        return self.major < other.major or (self.major == other.major and (self.minor < other.minor or patchcomp))
+
+    def __le__(self, other):
+        if not isinstance(other, K8sVersionInfo):
+            return False
+        patchcomp = self.minor == other.minor and self.patch <= other.patch
+        return self.major < other.major or (self.major == other.major and (self.minor < other.minor or patchcomp))
+
+    @classmethod
+    def extract_version(cls, string, separator="."):
+        components = string.strip().split(separator)
+        return cls(int(components[0]), int(components[1]), int(components[2]))
+
+    def check_for_version(self, major=None, minor=None, patch=None):
+        return (major is None or self.major == major) and \
+               (minor is None or self.minor == minor) and \
+               (patch is None or self.patch == patch)
+
+    def __str__(self):
+        return f"{self.major}.{self.minor}.{self.patch}"
+
+
+class CVEVersionInfo:
+    def __init__(self, lower_version, upper_version, less_than=False, equal=False):
+        self.lower_version = lower_version
+        self.upper_version = upper_version
+
+        self.less_than = less_than
+        self.equal = equal
+
+    def __eq__(self, other):
+        if not isinstance(other, CVEVersionInfo):
+            return False
+        return self.lower_version == other.lower_version and self.upper_version == other.upper_version and \
+            self.less_than == other.less_than and self.equal == self.equal
+
+    def is_version_affected(self, version_info):
+        if self.upper_version:
+            if self.lower_version:
+                gt = self.lower_version <= version_info
+            else:
+                gt = True
+            if self.less_than:
+                if self.equal:
+                    return gt and self.upper_version >= version_info
+                return gt and self.upper_version > version_info
+        else:
+            return self.lower_version == version_info
```

Comment:
I'm unfamiliar with the source data, but I'm just guessing I could be wrong. Should it be here lower or equal than just equal? Like if the CVE is not fixed in the next version?
```suggestion
            return self.lower_version <= version_info
```

Reply:
No this is correct as it is, for more information, check out the description/video/documentation [here](https://www.cve.org/AllResources/CveServices#cve-json-5)